### PR TITLE
Javascript javascript1 week2

### PR DESCRIPTION
**Pull request by SathiC1990**
Javascript week- 2 home work.

### CODE REVIEW

File: JS-week-2/candy_helper.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+let boughtCandy = [];
+
+function addCandy(candyType, weight){
+    let cadies = ['sweet','chocolate','toffee','chewing_gum'];
+    let candyPricePerGram = [0.5, 0.7, 0.3, 0.03];
+
+   // let candyIndex = cadies.findIndex((element) => element === candyType);
```
Comment (molefrog) [inline JS-week-2/candy_helper.js:7]: Using `findIndex` is very clever! I assume that you have commented it out to deal with cases when the candy type provided isn't found in the array. The documentation for this method says:

> The index of the first element in the array that passes the test. Otherwise, -1.

So, I can suggest adding an extra condition block to check if `candyIndex` is not `-1`, and if it is then simply return from the function. 

P.S. The solution you have provided below also works, so it's completely up to you which one to use.

File: JS-week-2/candy_helper.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+   let candyIndex;
+for (let index = 0; index < cadies.length; index++) {
+    if(cadies[index] === candyType){
+        candyIndex = index;
+        break;
+    }
+
+}
+
+    boughtCandy.push(candyPricePerGram[candyIndex] * weight);
```
Comment (molefrog) [inline JS-week-2/candy_helper.js:17]: If you decide to use the `for` approach, it's important to handle cases when `candyType` isn't valid. Let's see what happens if we call `addCandy("onion", 10)`:


```js
let candyIndex; // it starts with `undefined`
for (let index = 0; index < cadies.length; index++) {
    if(cadies[index] === candyType){
        candyIndex = index; // if candy type isn't in the array, this line will never run
        break;
    }
}

// so here: 
//   `candyIndex` is `undefined`
//   `candyPricePerGram[candyIndex]` is also `undefined`
//   `candyPricePerGram[candyIndex] * weight` is `NaN` 
boughtCandy.push(candyPricePerGram[candyIndex] * weight);
```

File: JS-week-2/weather_wear.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+function getWhattoWear (temInCelcius){
```
Comment (molefrog) [inline JS-week-2/weather_wear.js:1]: Just a small naming suggestion: when using [`camelCase` naming](https://developer.mozilla.org/en-US/docs/Glossary/Camel_case) the first letter of every word (except for the first one) should be capitalised. 

It's just a naming convention we (normally) use to name variables, functions etc (but of course, this can be project/team/company-specific). There are many other naming conventions, for example in Python or Ruby it is recommended to use `underscore_case`.

So, in your case this could be named:
```suggestion
function getWhatToWear (temInCelcius){
```

Comment (molefrog) [reply JS-week-2/weather_wear.js:1]: P.S. The function looks and works great!

File: JS-week-2/event_application.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+function weekdayCalc(days){
+    const weekdays = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+    let today = new Date();
+
+    today.setDate(today.getDate() + days);
+
+    console.log("The days is : "+weekdays[today.getDay()]);
+}
+
+weekdayCalc(8 );
```
Comment (molefrog) [inline JS-week-2/event_application.js:10]: Try adding more cases like negative numbers or zero to see how your program behaves. I think it should handle it well, since you are using the `setDate` method which is really robust, but it's always good to have more test cases to test against so called edge-cases. 

There is a [nice blog post](https://medium.com/jeremy-gottfrieds-tech-blog/algorithms-common-edge-cases-in-js-ce35a2d47674) on handling edge cases, it's a bit more advanced but I figured it might be interesting to read.

File: JS-week-2/candy_helper.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+let boughtCandy = [];
+
+function addCandy(candyType, weight){
+    let cadies = ['sweet','chocolate','toffee','chewing_gum'];
+    let candyPricePerGram = [0.5, 0.7, 0.3, 0.03];
```
Comment (molefrog) [inline JS-week-2/candy_helper.js:5]: An alternative way to store this data is to introduce an object like this:

```js
let candies = {
    'sweet': 0.5,
    'chocolate': 0.7,
    'toffee': 0.3,
    'chewing_gum': 0.03
};
```

This way, to get the price per gram of a specific candy you can write `candies['sweet']`. 

P.S. There is nothing wrong with your solution, and I think you should keep it as is because you already have the code for locating the element in these arrays written. Just wanted to bring up another method of storing data for this exercise 😉

File: JS-week-2/weather_wear.js (hunk 3 above)
Comment (SathiC1990) [reply JS-week-2/weather_wear.js:1]: Done. Sorry for the late.

File: JS-week-2/event_application.js (hunk 4 above)
Comment (SathiC1990) [reply JS-week-2/event_application.js:10]: Hi @molefrog I have added the more cases, and code is working.
